### PR TITLE
Fix #1314 event in onAddClick is no longer mandatory - with test

### LINKED ISSUE #1314: ArrayFieldTemplate - onAddClick should not require an argument
### Prerequisites

- [ ] I have read the [documentation](https://react-jsonschema-form.readthedocs.io/);
- [ ] In the case of a bug report, I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [ ] Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net/n1k0/f2y3fq7L/6/) or a [shared playground link](https://mozilla-services.github.io/react-jsonschema-form/) demonstrating the issue.

### Description

Right now calling this.props.onAddClick() from an ArrayFieldTemplate gives an error because it expects you to pass in an event as an argument. I don't see why passing in an event should be required so we should make it optional.

### PR DESCRIPTION
**Pull request by Parro**
### Reasons for making this change

Calling `onAddClick` from a custom `ArrayFieldTemplate` gives error if `event` is not passed as parameter, but there are several cases in which `event` is not set. This PR makes `event` optional fixing #1314. Code update taken from #1365 and #1324.

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [x] **I'm adding or updating code**
  - [x] I've added and/or updated tests
  - [x] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

File: docs/advanced-customization.md [hunk 1]
```diff
@@ -120,7 +120,7 @@ The following props are passed to each `ArrayFieldTemplate`:
 - `disabled`: A boolean value stating if the array is disabled.
 - `idSchema`: Object
 - `items`: An array of objects representing the items in the array. Each of the items represent a child with properties described below.
-- `onAddClick: (event) => void`: A function that adds a new item to the array.
+- `onAddClick: ([event]) => void`: A function that adds a new item to the array. `event` is optional.
```
Comment (epicfaace) [inline docs/advanced-customization.md:123]:
```suggestion
- `onAddClick: (event?) => void`: A function that adds a new item to the array. `event` is optional.
```

File: packages/core/test/ArrayField_test.js [hunk 2]
```diff
@@ -249,6 +273,18 @@ describe("ArrayField", () => {
...
 
+    it("should add a field when clicking add button even if event is not passed to onAddClick", () => {
+      const { node } = createFormComponent({
+        schema,
+        ArrayFieldTemplate: CustomOnAddClickTemplate,
+      });
+
+      Simulate.click(node.querySelector(".array-item-add button"));
+
+      expect(node.querySelector(".array-item").hasAttribute(ArrayKeyDataAttr))
```
Comment (epicfaace) [inline packages/core/test/ArrayField_test.js:284]: Can you instead make the test check just that `.array-item` exists? I don't think we need to check if the key is present, so we should safely be able to remove that check and the data-rjsf-itemkey attribute (which was only a hack needed to check the keys of array items for specific tests regarding array reordering)

Comment (Parro) [reply packages/core/test/ArrayField_test.js]: I fixed the test but I've never used chai... Did i used the right syntax?

**Comment by Parro** [comment]
> Thanks! Can you also update the types here: https://github.com/rjsf-team/react-jsonschema-form/blob/master/packages/core/index.d.ts#L184

Updated, I also removed the second event params, there is no currying for this function 😉

File: packages/core/test/ArrayField_test.js [hunk 3]
```diff
@@ -62,6 +62,30 @@ const ExposedArrayKeyTemplate = function(props) {
...
 
+const CustomOnAddClickTemplate = function(props) {
+  return (
+    <div className="array">
+      {props.items &&
+        props.items.map(element => (
+          <div
+            key={element.key}
+            className="array-item"
+            data-rjsf-itemkey={element.key}>
```
Comment (epicfaace) [inline packages/core/test/ArrayField_test.js:73]: Can you remove this attribute? then we should be good

Comment (Parro) [reply packages/core/test/ArrayField_test.js]: Fixed

**Pull request merged**